dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -10,49 +10,47 @@ def update_last_confirmation_date
     head :no_content
   end
 
-  # PATCH /api/v1/users_school_infos/<id>/update_end_date
-  def update_end_date
-    ActiveRecord::Base.transaction do
-      @user_school_info.update!(end_date: DateTime.now, last_confirmation_date: DateTime.now)
-      @user_school_info.user.update!(properties: {last_seen_school_info_interstitial: DateTime.now})
-    end
-
-    head :no_content
-  end
-
-  # PATCH /api/v1/user_school_infos/<id>/update_school_info_id
-  # A new row is added to the user school infos table when a teacher updates current school to a different school,
-  # then the school_info_id is updated in the users table.
-
-  def update_school_info_id
-    ActiveRecord::Base.transaction do
-      school = @user_school_info.school_info.school
-      unless school
-        school = School.create!(new_school_params)
-        school.school_info.create!(school_info_params)
+  # PATCH /api/v1/users_school_infos/<id>/school_info
+  def update
+    if school_info_params[:school_id].present?
+      existing_school_info = current_user.school_info
+      if existing_school_info && !existing_school_info.complete?
+        existing_school_info.update!(school_info_params)
```

Comment:
I'm still concerned this will have unintended affects on other users. Can you test what happens if two users share the same school info then one changes it?

Reply:
I think this is, somewhat, an invented situation.  From production:

```
mysql> select school_infos.id, count(*) from users join school_infos on users.school_info_id = school_infos.id having count(*) > 1 limit 10;
+----+----------+
| id | count(*) |
+----+----------+
| 34 |   668864 |
+----+----------+
1 row in set (1.92 sec)
```

It looks like there's exactly one `SchoolInfo` that's shared by more than one user - and that one is shared by more than 600,000 teachers, so it's probably an edge case.

Looking at that one record, it definitely counts as 'incomplete' so we might worry about one of these teachers updating their school info and having it implicitly applied to all the others.
```
mysql> select * from school_infos where id=34;
+----+---------+-------------+------+-------+--------------------+-----------------------+----------------------+-----------+--------------+-------------+--------------+---------------------+---------------------+-----------------+
| id | country | school_type | zip  | state | school_district_id | school_district_other | school_district_name | school_id | school_other | school_name | full_address | created_at          | updated_at          | validation_type |
+----+---------+-------------+------+-------+--------------------+-----------------------+----------------------+-----------+--------------+-------------+--------------+---------------------+---------------------+-----------------+
| 34 | NULL    | public      | NULL | CA    |               NULL |                     1 | NULL                 | NULL      |         NULL | NULL        | NULL         | 2016-08-29 22:05:29 | 2016-11-19 22:05:05 | full            |
+----+---------+-------------+------+-------+--------------------+-----------------------+----------------------+-----------+--------------+-------------+--------------+---------------------+---------------------+-----------------+
1 row in set (0.01 sec)

```

Reply:
(Earlier comment deleted) You're right, this is a concern:
```
mysql> select school_infos.id, count(*) from users join school_infos on users.school_info_id = school_infos.id group by school_infos.id having count(*) > 1 order by count(*) desc limit 10;
+--------+----------+
| id     | count(*) |
+--------+----------+
|   7220 |    53960 |
| 110676 |    14765 |
| 158025 |     7064 |
| 108735 |     4898 |
| 198506 |     3743 |
|   8626 |     2046 |
| 158026 |     1803 |
|   8479 |     1721 |
| 158084 |     1334 |
| 158202 |     1213 |
+--------+----------+
```

Reply:
Possible solutions:
- We should never update a partial school info in place.  We might want to notice and delete unused school infos in this case.
- We should only update a partial school info in place if it _only_ belongs to you.  Otherwise, we should make a new one.

Reply:
Maybe relevant: About 60,000 SchoolInfos are not currently attached to any User record:
```
mysql> select count(*) from school_infos left outer join users on users.school_info_id = school_infos.id where users.id is null;
+----------+
| count(*) |
+----------+
|    59586 |
+----------+
1 row in set (0.92 sec)
```